great_expectations/core/expectation_suite.py

```diff
@@ -669,29 +672,204 @@ def add_expectation(
+        self,
+    ) -> List[ExpectationConfiguration]:
+        """
+        Returns "ExpectationConfiguration" list, grouped by "expectation_type", in predetermined designated order.
+        """
+        table_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_table_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        column_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_column_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        column_pair_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_column_pair_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        multicolumn_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_multicolumn_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        return (
+            table_expectation_configurations
+            + column_expectation_configurations
+            + column_pair_expectation_configurations
+            + multicolumn_expectation_configurations
+        )
+
     def get_table_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of table expectations."""
-        return [
-            e
-            for e in self.expectations
-            if e.expectation_type.startswith("expect_table_")
-        ]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type() == MetricDomainTypes.TABLE,
+                self.expectations,
+            )
+        )
+
+        expectation_configuration: ExpectationConfiguration
+        for expectation_configuration in expectation_configurations:
+            expectation_configuration.kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+
+        return expectation_configurations
 
     def get_column_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of column map expectations."""
-        return [e for e in self.expectations if "column" in e.kwargs]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type() == MetricDomainTypes.COLUMN,
+                self.expectations,
+            )
+        )
 
+        expectation_configuration: ExpectationConfiguration
+        kwargs: dict
+        column_name: str
+        for expectation_configuration in expectation_configurations:
+            kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+            column_name = kwargs.pop("column")
```

Comment:
Could we replace lines 818 and 819 with `expectation_configuration.kwargs = kwargs`? I'm not sure why we pop `column` off then pass it in explicitly?

Reply:
@billdirks This is designed to make it appear as the first key.

Reply:
I see. Thanks!